File: frontend/simda/lib/models/CommentDto.dart
import 'package:simda/models/UserDto.dart';

class CommentDto {
  final List<CommentDto>? cCommentList;
  final int cmtId;
  final String content;
  final int feedId;
  final int? pcmtId;
  final String regTime;
  final UserDto userDto;

  CommentDto({
    this.cCommentList,
    required this.cmtId,
    required this.content,
    required this.feedId,
    this.pcmtId,
    required this.regTime,
    required this.userDto
  });

  factory CommentDto.fromJson(Map<String, dynamic> json) {
    // print((json['cCommentList'] as List<dynamic>?)
    //     ?.map((e) => CommentDto.fromJson(e))
    //     .toList());
    // print((json['ccommentList'] as List<dynamic>?)
    //     ?.map((e) => CommentDto.fromJson(e))
    //     .toList());
    // print((json['commentList'] as List<dynamic>?)
    //     ?.map((e) => CommentDto.fromJson(e))
    //     .toList());
    return CommentDto(
      content: json['content'],
      cmtId: json['cmtId'],
      cCommentList: (json['ccommentList'] as List<dynamic>?)
          ?.map((e) => CommentDto.fromJson(e))
          .toList(),
      feedId: json['feedId'],
      pcmtId: json['pcmtId'],
      regTime: json['regTime'],
      userDto: UserDto.fromJson(json['userDto']),
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'cCommentList': cCommentList??[],
      'cmtId': cmtId,
      'content': content,
      'feedId': feedId,
      'pcmtId': pcmtId,
      'regTime': regTime,
      'userDto': userDto.toJson(),
    };
  }
}

File: ProjectSimda/simda/src/main/java/ssafy/a709/simda/dto/CommentDto.java
package ssafy.a709.simda.dto;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import ssafy.a709.simda.domain.Comment;

import java.sql.Timestamp;
import java.text.SimpleDateFormat;
import java.util.ArrayList;
import java.util.List;
import java.util.TimeZone;

@Builder
@AllArgsConstructor
@NoArgsConstructor
@Data
public class CommentDto {

    // Comment Id
    private int cmtId;

    private UserDto userDto;

    private int feedId;

    private int pCmtId;

    private String regTime;

    // Content
    private String content;
    private List<CommentDto> cCommentList;

    // Comment(Entity)를 CommentDto로 변환
    public static CommentDto changeToCommentDto(Comment comment){
        return CommentDto.builder()
                .cmtId(comment.getCmtId())
                .userDto(UserDto.changeToUserDto(comment.getUser()))
                .feedId(comment.getFeed().getFeedId())
                .pCmtId(comment.getPComment() == null ? -1 : comment.getPComment().getCmtId())
                .regTime(convertUtcToSeoul(comment.getRegTime()))
                .content(comment.getContent())
                .build();
    }

    private static String convertUtcToSeoul(Timestamp regTime){
        SimpleDateFormat sdf = new SimpleDateFormat("yyyy-MM-dd HH:mm:ss");
        sdf.setTimeZone(TimeZone.getTimeZone("Asia/Seoul")); // UTC+9 타임존 설정

        return sdf.format(regTime);
    }
}

